.travis.yml

```diff
@@ -43,20 +43,22 @@ matrix:
     - php: hhvm
     - php: nightly
 
-before_script:
+before_install:
   - (phpenv config-rm xdebug.ini || exit 0)
   - mkdir -p ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d && echo "memory_limit=-1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
   - composer selfupdate
   - composer config -q -g github-oauth.github.com $GITHUB_OAUTH_TOKEN
-  - composer global require phpunit/phpunit:@stable fabpot/php-cs-fixer --no-update
-  - composer global update --prefer-dist --no-interaction
+  - if [ "$TARGET" = "cs_dry_run" ]; composer global require fabpot/php-cs-fixer --no-update; fi;
+  - if [ "$TARGET" = "test" ]; composer global require phpunit/phpunit:@stable --no-update; fi;
   - if [ "$SYMFONY_VERSION" != "" ]; then composer require "symfony/symfony:${SYMFONY_VERSION}" --no-update; fi;
-  - travis_wait composer update --prefer-dist --no-interaction $COMPOSER_FLAGS
   - export PATH=$HOME/.local/bin:$PATH
-  - pip install -r Resources/doc/requirements.txt --user `whoami`
 
-script:
- - make $TARGET
+install:
+  - if [ "$TARGET" != "docs" ]; composer global update --prefer-dist --no-interaction; fi;
+  - if [ "$TARGET" = "test" ]; travis_wait composer update --prefer-dist --no-interaction $COMPOSER_FLAGS; fi;
+  - if [ "$TARGET" = "docs" ]; pip install -r Resources/doc/requirements.txt --user `whoami`; fi;
```

Comment:
I prefer it this way, as the 2 actions are not really related. It's more like: If it's "test" or "cs-fix", run a global update. If it's docs run a pip install.

As `if [[ "$TARGET" = "test" || "$TARGET" = "cs_dry_run" ]]; then ...; fi;` is more complex than `if [ "$TARGET" != "docs" ]; then ...; fi;`.